src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts

```diff
@@ -37,53 +45,85 @@ export class AutofillProjectFormComponent implements OnInit {
       Validators.required,
       Validators.pattern(/^10.\d{4,9}\/[-._;()\/:A-Z0-9]+$/i)
     ]));
+    this.geoAccessionCtrl = new FormControl('', Validators.compose([
+      Validators.required,
+      Validators.pattern(/^GSE.*$/i)
+    ]));
     this.projectInCache$ = from(this.projectCacheService.getProject());
   }
 
-  showError(control: FormControl): string {
+  showError(control: FormControl, message: string): string {
     if (control.touched && control.errors) {
       const errors = control.errors;
 
       if (errors['required']) {
         return 'This field is required';
       }
       if (errors['pattern']) {
-        return 'The DOI must be a valid DOI. E.g. 10.1038/s41467-021-26902-8'
+        return message;
       }
     }
   }
 
   submitForm() {
     this.alertService.clear();
-    if (this.publicationDoiCtrl.invalid) {
+
+    if (this.publicationDoiCtrl.invalid && this.geoAccessionCtrl.invalid) {
       this.publicationDoiCtrl.markAsTouched();
+      this.geoAccessionCtrl.markAsTouched();
       return;
     }
 
-    if (this.publicationDoiCtrl.value) {
-      const doi = this.publicationDoiCtrl.value;
-      this.loading = true;
-      forkJoin({
-        projects: this.getProjectsWithDOI(doi),
-        doiExists: this.doesDoiExist(doi)
-      }).subscribe(({projects, doiExists}) => {
-          this.showExistingProjectsAlert(projects);
-
-          if (!doiExists) {
-            this.showDOINotExistsAlert(doi);
-          }
-
-          if (doiExists && projects.length == 0) {
-            this.createProject(doi);
-          }
-
-          this.loading = false;
-        },
-        error => {
-          this.alertService.error('An error occurred', error.message);
-          this.loading = false;
-        });
+    const doi = this.publicationDoiCtrl.value;
+    if (this.hasDoi && doi) {
+      this.importProjectUsingDoi(doi);
     }
+
+    const geoAccession = this.geoAccessionCtrl.value;
+    if (this.hasGeo && geoAccession) {
+      this.importProjectUsingGeo(geoAccession);
+    }
+  }
+
+  private importProjectUsingDoi(doi: string) {
+    this.loading = true;
+    forkJoin({
+      projects: this.autofillProjectService.getProjectsWithDOI(doi),
+      doiExists: this.autofillProjectService.doesDoiExist(doi)
+    }).subscribe(({projects, doiExists}) => {
+        this.showExistingProjectsAlert(projects, 'doi');
+
+        if (!doiExists) {
+          this.showDOINotExistsAlert(doi);
+        }
+
+        if (doiExists && projects.length == 0) {
+          this.createProjectWithDoi(doi);
+        }
+
+        this.loading = false;
+      },
+      error => {
+        this.alertService.error('An error occurred', error.message);
+        this.loading = false;
+      });
+  }
+
+  private importProjectUsingGeo(geoAccession) {
+    this.loading = true;
+    forkJoin({
+      projects: this.autofillProjectService.getProjectsWithGeo(geoAccession),
```

Comment:
Why forkjoin? There's no combination going on - you can just subscribe to the result of getProjectWithGeo

Reply:
wonder why that too

Reply:
yeah, it's probably my copy-paste deed from geo, fixing...

Reply:
fixed